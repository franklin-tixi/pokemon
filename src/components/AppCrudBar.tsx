import { AppButton } from "./button/AppButton";
import { AppTextField } from "./inputs/AppTextField";
import { LabelCrud } from "./../helpers/const/global";

/**
 * Properties
 */
interface Props {
  title?: string;
  onClickAdd?: () => void;
  onSearch?: (value: string) => void;
}

/**
 * Component that contains the search engine and new button
 * @param props
 * @returns
 */
export function AppCrudBar(props: Props) {
  return (
    <>
      {/* TITLE */}
      <h3>{props.title}</h3>
      <div style={{ display: "flex", justifyContent: "space-between" }}>
        {/* INPUT SEARCH */}
        <AppTextField
          ariaLabel="input-search"
          className="search text-field"
          placeholder="Buscar"
          onChange={props.onSearch}
          name="search"
          type="text"
        />
        {/* BUTTON NEW */}
        <AppButton
          className="btn-new"
          label={LabelCrud.NEW}
          onClick={props.onClickAdd}
        />
      </div>
    </>
  );
}
