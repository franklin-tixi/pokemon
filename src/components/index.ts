/**
 * Barrel for components
 */

/** COMMON */
export * from "./AppCrudBar";

/** BUTTON */
export * from "./button/AppButton";

/** DIALOG */
export * from "./dialog/AppFormDialog";

/** INPUTS */
export * from "./inputs/AppTextField";

/** SLIDER */
export * from "./slider/AppSliderBar";

/** TABLE */
export * from "./table/AppTable";
