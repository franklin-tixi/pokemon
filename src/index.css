/********************************************************
*                           ROOT                        *
********************************************************/
:root {
  --main-color: #262626;
  --main-border-color: #b3b3b3;
  --main-border-color-primary: #197ccc;
  --main-border-radius: 3px;
  --main-bg-color: #fff;
  --main-bg-color-primary: #248ce4;
  font-family: Inter, Avenir, Helvetica, Arial, sans-serif;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  color: var(--main-color);
  background-color: var(--main-bg-color);
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
  margin: 20px 40px 40px 40px;
}

/********************************************************
*                         TABLE                         *
********************************************************/
table,
tr,
td {
  border-collapse: collapse;
}

td {
  border-collapse: collapse;
  padding: 5px 40px;
  text-align: center;
}

thead {
  font-weight: 600;
}

/********************************************************
*                    BEAUTY SCROLL BAR                  *
********************************************************/
@media (min-width: 100px) {
  ::-webkit-scrollbar {
    width: 11px;
    height: 11px;
  }

  ::-webkit-scrollbar-track {
    background: rgba(224, 224, 224, 1);
  }

  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background: #f5f5f5;
    background-color: #d9d9d9;
    box-shadow: inset 0 0 8px rgba(12, 9, 9, 0.5) !important;
  }
}

/********************************************************
*                         ICON                          *
********************************************************/
i.edit {
  background-image: url(../../../src/assets/images/icons/edit.png);
  background-repeat: no-repeat;
  background-size: 18px;
  background-position-x: 15%;
  background-position-y: 50%;
  padding-right: 20px;
  padding-left: 20px;
}

i.delete {
  background-image: url(../../../src/assets/images/icons/delete.png);
  background-repeat: no-repeat;
  background-size: 18px;
  background-position-x: 15%;
  background-position-y: 50%;
  padding-right: 20px;
  padding-left: 20px;
}

i:hover {
  cursor: pointer;
}
