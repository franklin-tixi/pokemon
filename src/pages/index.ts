/**
 * Barrel for Pokemon
 */
export * from "./PokemonModel";
export * from "./PokemonService";
export * from "./PokemonPage";
