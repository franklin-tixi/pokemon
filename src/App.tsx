import "./App.css";
import { PokemonPage } from "./pages/PokemonPage";
function App() {
  return (
    <>
      <PokemonPage />
    </>
  );
}

export default App;
