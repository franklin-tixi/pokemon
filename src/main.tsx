import React from "react";
import ReactDOM from "react-dom/";
import App from "./App";
import "./index.css";
import "./components/button/AppButton.css";
import "./components/dialog/AppFormDialog.css";
import "./components/inputs/AppTextField.css";
import "./components/table/AppTable.css";

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById("root")
);
