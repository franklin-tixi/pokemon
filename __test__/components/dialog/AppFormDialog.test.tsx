import { render, screen } from "@testing-library/react";
import React from "react";
import { AppFormDialog } from "../../../src/components/";

describe("Pruebas AppFormDialog", () => {
  const title = "Titulo Pokemon";
  const open: boolean = true;

  test("Debe realizar match con el snapshot", () => {
    const mockCallBack = jest.fn();
    const { container } = render(
      <AppFormDialog onClose={mockCallBack} open={open} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar en titulo - Titulo Pokemon", () => {
    const mockCallBack = jest.fn();
    render(<AppFormDialog onClose={mockCallBack} title={title} open={open} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe de mostrar en titulo - en un h4", () => {
    const mockCallBack = jest.fn();
    render(<AppFormDialog onClose={mockCallBack} title={title} open={open} />);
    expect(screen.getByRole("heading", { level: 4 })).toBeTruthy();
  });

  test("Debe contener 2 botones", () => {
    const mockCallBack = jest.fn();
    render(<AppFormDialog onClose={mockCallBack} title={title} open={open} />);
    expect(screen.getAllByRole("button").length).toBe(2);
  });
});
