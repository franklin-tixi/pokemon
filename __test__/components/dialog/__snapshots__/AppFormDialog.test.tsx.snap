// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas AppFormDialog Debe realizar match con el snapshot 1`] = `
<div>
  <div
    class="form-crud"
  >
    <form
      aria-label="form"
    >
      <h4
        style="text-align: center;"
      />
      <div
        style="margin: 2rem; display: flex; justify-content: center;"
      />
      <div
        style="margin-top: 2rem; margin-bottom: 2rem; display: flex; justify-content: center;"
      >
        <div>
          <button
            class="save"
            style="margin-left: 1rem;"
            type="submit"
          >
            Guardar
          </button>
        </div>
        <div>
          <button
            class="cancel"
            style="margin-left: 1rem;"
          >
            Cancelar
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
